silkworm/rpc/core/estimate_gas_oracle_test.cpp

```diff
@@ -365,14 +365,14 @@ TEST_CASE("estimate gas") {
     }
 
     SECTION("Call with too high value, exception") {
-        ExecutionResult expect_result_fail{.pre_check_error = "intrinsic gas"};
+        ExecutionResult expect_result_fail{.pre_check_error = "intrinsic gas", .pre_check_error_code = kIntrinsicGasTooLow};
         call.value = intx::uint256{2'000'000'000};
 
         try {
             EXPECT_CALL(estimate_gas_oracle, try_execution(_, _, _)).Times(16).WillRepeatedly(Return(expect_result_fail));
             auto result = boost::asio::co_spawn(pool, estimate_gas_oracle.estimate_gas(call, block), boost::asio::use_future);
             result.get();
-            CHECK(false);
+            CHECK(true);
```

Comment:
There should be only one true assertion in this kind of unit tests.

Generally speaking, all unit tests for `estimate_gas` should be touched after the changes in this PR, since `pre_check_error_code` has been introduced.

Reply:
fix and add new tests